bpf/process/bpf_generic_lsm.c

```diff
@@ -154,6 +168,75 @@ __attribute__((section("lsm/5"), used)) int
 generic_lsm_output(void *ctx)
 {
 	generic_output(ctx, (struct bpf_map_def *)&process_call_heap, MSG_OP_GENERIC_LSM);
+#ifdef __LARGE_MAP_KEYS
+	struct msg_generic_kprobe *e;
+	int zero = 0;
+
+	e = map_lookup_elem(&process_call_heap, &zero);
+	if (e && e->common.flags & MSG_COMMON_FLAG_IMA_HASH)
+		tail_call(ctx, &lsm_calls, TAIL_CALL_IMA_HASH);
```

Comment:
Isn't there a race with this approach?
That is, what guarantees are there that user-space will do the map lookup before the entry is added?

Reply:
Can we tailcall to a program that does the `generic_output`  from an lsm.s program?

Reply:
> Can we tailcall to a program that does the `generic_output` from an lsm.s program?

Unfortunately, no(. Becase we can't use `BPF_MAP_TYPE_PROG_ARRAY` maps in lsm.s programs. We can tail call in, but we can't taill call out :).